<launch>
  <!-- Launch the RealSense camera -->
  <include file="$(find realsense2_camera)/launch/rs_camera.launch">
    <!-- RealSense camera configuration parameters -->
    <arg name="enable_pointcloud" value="true" />
    <arg name="align_depth" value="true" />
    <arg name="color_width" value="848" />
    <arg name="color_height" value="480" />
    <arg name="color_fps" value="30" />
    <arg name="depth_width" value="848" />
    <arg name="depth_height" value="480" />
    <arg name="depth_fps" value="30" /> 
  </include>

  <arg name="pkg_path" value="$(find vader_perception)" />
  
  <!-- Then set the weights_path parameter with the resolved path -->
  <param name="fruit_weights_path" value="$(arg pkg_path)/weights/orientation2.pt" />
  <param name="fruit_drive_url" value="https://drive.google.com/uc?id=1jgSxQraHAupce1RNkpaMzL7MUVqtjHZG" />
  <param name="peduncle_weights_path" value="$(arg pkg_path)/weights/peduncle_2.pt" />
  <param name="peduncle_drive_url" value="https://drive.google.com/uc?id=1PVYVIw1jzbX4jpGHCnGwGPXbqa2IcJJv" />
  
  
  <!-- Start the fruit detection node -->
  <node name="pose_estimation" pkg="vader_perception" type="pose_estimation_node.py" output="screen">
    <param name="fruit_confidence" value="0.8" />
    <param name="peduncle_confidence" value="0.7" />
  </node>

  <!-- Launch RViz with custom configuration -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg pkg_path)/rviz_config/gripper_cam.rviz" />
</launch>
